client/settings/fraud-protection/advanced-settings/index.js

```diff
@@ -0,0 +1,199 @@
+import AddressMismatchRuleCard from './cards/address-mismatch';
+import OrderVelocityRuleCard, {
+	OrderVelocityValidation,
+} from './cards/order-velocity';
+import PurchasePriceThresholdRuleCard, {
+	PurchasePriceThresholdValidation,
+} from './cards/purchase-price-threshold';
+import OrderItemsThresholdRuleCard, {
+	OrderItemsThresholdValidation,
+} from './cards/order-items-threshold';
+import FraudPreventionSettingsContext from './context';
+import { useSettings } from '../../../data';
+import ErrorBoundary from 'wcpay/components/error-boundary';
+import { getAdminUrl } from 'wcpay/utils';
+
+const Breadcrumb = () => (
+	<h2 className="fraud-protection-header-breadcrumb">
+		<Link
+			type="wp-admin"
+			href={ getAdminUrl( {
+				page: 'wc-settings',
+				tab: 'checkout',
+				section: 'woocommerce_payments',
+			} ) }
+		>
+			{ __( 'WooCommerce Payments', 'woocommerce-payments' ) }
+		</Link>
+		&nbsp;&gt;&nbsp;
+		{ __( 'Advanced fraud protection', 'woocommerce-payments' ) }
+	</h2>
+);
+
+const SaveFraudProtectionSettingsButton = ( { children } ) => {
+	const headerElement = document.querySelector(
+		'.woocommerce-layout__header-wrapper'
+	);
+	return headerElement && ReactDOM.createPortal( children, headerElement );
+};
+
+const FraudProtectionAdvancedSettingsPage = () => {
+	const { settings, saveSettings, isLoading } = useSettings();
+	const [ isSavingSettings, setIsSavingSettings ] = useState( false );
+	const [ validationError, setValidationError ] = useState( null );
+	const [
+		advancedFraudProtectionSettings,
+		setAdvancedFraudProtectionSettings,
+	] = useState( {} );
+
+	useEffect( () => {
+		setAdvancedFraudProtectionSettings(
+			settings.advanced_fraud_protection_settings
+		);
+	}, [ settings ] );
+
+	useLayoutEffect( () => {
+		const saveButton = document.querySelector(
+			'.fraud-protection-header-save-button'
+		);
+		if ( saveButton ) {
+			document
+				.querySelector( '.woocommerce-layout__header-heading' )
+				.after( saveButton );
+		}
+	} );
+
+	const validateSettings = ( fraudProtectionSettings ) => {
+		setValidationError( null );
+		let validationResult = true;
+		validationResult &&= OrderItemsThresholdValidation(
+			fraudProtectionSettings,
+			setValidationError
+		);
+		validationResult &&= OrderVelocityValidation(
+			fraudProtectionSettings,
+			setValidationError
+		);
+		validationResult &&= PurchasePriceThresholdValidation(
+			fraudProtectionSettings,
+			setValidationError
+		);
+		return validationResult;
+	};
+
+	const handleSaveSettings = async () => {
+		if ( validateSettings( settings.advanced_fraud_protection_settings ) ) {
+			setIsSavingSettings( true );
+			await saveSettings( settings );
+			setIsSavingSettings( false );
+		} else {
+			window.scrollTo( {
+				top: 0,
+			} );
+		}
+	};
+
+	// Hack to make "WooCommerce > Settings" the active selected menu item.
+	const wcSettingsMenuItem = document.querySelector(
```

Comment:
Can this be moved to a `useEffect` hook?

Reply:
Thanks for suggesting this, didn't know that needed to be done in an `useEffect` wrapper.

Reply:
It's just to prevent running it multiple times during the component lifespan. As the component contains a state, it will most likely re-render multiple times, and having this logic in the component body will make it run multiple times as well.